tools/yamcan/templates/message_unpack.mako

```diff
@@ -101,12 +101,14 @@
     tmp_${signal.name} = (tmp_${signal.name} & ${(2**(signal.native_representation.bit_width - 1) - 1)}U) * (((tmp_${signal.name} & (1U << ${(signal.native_representation.bit_width - 1)})) != 0U) ? -1.0f : 1.0f);
     %endif
     %if "float" in signal.datatype.value:
-    sigrx->${sig_name} = (${signal.datatype.value})(tmp_${signal.name}) * ${float(signal.scale)}f + (${float(signal.offset)}f);
+    sigrx->${sig_name} = (${signal.datatype.value})(
+        ((${signal.datatype.value})tmp_${signal.name}) * ${float(signal.scale)}f
+        + ${float(signal.offset)}f
+    );
     %elif "int" in signal.datatype.value:
     sigrx->${sig_name} = (tmp_${signal.name} * ${int(signal.scale)}) + (${int(signal.offset)});
     %endif
 %else:
-    (void)m;
-    (void)val;
```

Comment:
Why did this diff? If youre in this case, you should need to disregard both function args, maybe im missing something?